src/ws_transport.ts

```diff
@@ -81,7 +86,29 @@ export class WsTransport implements Transport {
     this.options = options;
     const u = server.src;
     this.encrypted = u.indexOf("wss://") === 0;
-    this.socket = new WebSocket(u);
+    const { fetch, useHTTPUpgrade } = options;
+    if (useHTTPUpgrade && fetch) {
```

Comment:
This upgrade mechanism is very specific to cloudflare - there's a header, and use of fetch etc.
I think it would be more useful and flexible, to simply provide a function that takes returns the WebSocket based on the input URLs, and instead contribute an implementation of this function (to this library) that could be provided as a connection argument. We can then properly expose this to the base library (albeit WS specific), so that there isn't any sort of mismatch from what the base library is expecting.

imho - the API should be: 
`export type WsSocketFactory = (u: string, opts: unknown) => Promise<{socket: Websocket, encrypted: boolean}>`